app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,31 @@ def show
     @user = @post.user
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @post.likescounter = 0
+    @post.commentscounter = 0
+    @post.update_post_counter
+
+    respond_to do |format|
+      format.html do
+        if @post.save
+          flash[:success] = 'Post saved successfully'
+          redirect_to new_user_post_path(current_user)
+        else
+          flash.now[:error] = 'Error: Post could not be saved. Please try again.'
+          render :new
+        end
+      end
+    end
```

Comment:
- [ ] After creating a post, it would be nicer if the user is directed to the post-show page. I suggest updating the path at line 27 to the path of the post-show page.

Reply:
Thanks a lot for your review let me work on the changes